frontend/src/metabase/admin/permissions/selectors/confirmations.ts

```diff
@@ -123,20 +131,45 @@ export function getControlledDatabaseWarningModal(
   entityId: EntityId,
 ) {
   if (currDbPermissionValue !== DataPermissionValue.CONTROLLED) {
-    const [entityType, entityTypePlural] = isTableEntityId(entityId)
-      ? [t`table`, t`tables`]
-      : isSchemaEntityId(entityId)
-      ? [t`schema`, t`schemas`]
-      : [t`entity`, t`entities`];
+    const [entityType, entityTypePlural] = getEntityTypeFromId(entityId);
     return {
-      title: t`Change access to this database to granular?`,
+      title: t`Change access to this database to “Granular”?`,
       message: t`Just letting you know that changing the permission setting on this ${entityType} will also update the database permission setting to “Granular” to reflect that some of the database’s ${entityTypePlural} have different permission settings.`,
       confirmButtonText: t`Change`,
       cancelButtonText: t`Cancel`,
     };
   }
 }
 
+export function getWillRevokeNativeAccessWarningModal(
+  permissions: GroupsPermissions,
+  groupId: number,
+  entityId: EntityId,
+) {
+  // if the db is set to query builder and native for this group
+  // then warn the user that the change will downgrade native permissions
+  const currDbCreateQueriesPermission = getSchemasPermission(
+    permissions,
+    groupId,
+    { databaseId: entityId.databaseId },
+    DataPermission.CREATE_QUERIES,
+  );
+
+  if (
+    currDbCreateQueriesPermission ===
+    DataPermissionValue.QUERY_BUILDER_AND_NATIVE
+  ) {
+    const [entityType] = getEntityTypeFromId(entityId);
+
+    return {
+      title: t`Change access to this database to “Granular”?`,
+      message: t`As part of providing granular permissions for this one ${entityType}, we will also have to remove this group's native querying permissions from all tables and schemas in this database.`,
+      confirmButtonText: t`Change`,
```

Comment:
To disambiguate, for translators:

```suggestion
      confirmButtonText: c("This is a verb, for a confirmation button").t`Change`,
```